filter_plugins/oo_filters.py

```diff
@@ -273,16 +276,52 @@ def oo_filter_list(data, filter_attr=None):
                 returns [ { a: 1, b: True },
                           { a: 5, b: True } ]
         '''
-        if not issubclass(type(data), list):
+        if not isinstance(data, list):
             raise errors.AnsibleFilterError("|failed expects to filter on a list")
 
-        if not issubclass(type(filter_attr), str):
-            raise errors.AnsibleFilterError("|failed expects filter_attr is a str")
+        if not isinstance(filter_attr, basestring):
+            raise errors.AnsibleFilterError("|failed expects filter_attr is a str or unicode")
 
         # Gather up the values for the list of keys passed in
         return [x for x in data if x.has_key(filter_attr) and x[filter_attr]]
 
     @staticmethod
+    def oo_nodes_with_label(nodes, label=None, value=None):
+        ''' Filters a list of nodes by label '''
+        if not isinstance(nodes, list):
+            raise errors.AnsibleFilterError("failed expects to filter on a list")
+        if label is None:
+            return nodes
+        elif not isinstance(label, basestring):
+            raise errors.AnsibleFilterError("failed expects label to be a string")
+        if value is not None and not isinstance(value, basestring):
+            raise errors.AnsibleFilterError("failed expects value to be a string")
+
+        def label_filter(node):
+            ''' filter function for testing if node should be returned '''
+            if not isinstance(node, dict):
+                raise errors.AnsibleFilterError("failed expects to filter on a list of dicts")
+            if 'openshift_node_labels' in node:
+                labels = node['openshift_node_labels']
+            elif 'cli_openshift_node_labels' in node:
+                labels = node['cli_openshift_node_labels']
+            elif 'openshift' in node and 'node' in node['openshift'] and 'labels' in node['openshift']['node']:
+                labels = node['openshift']['node']['labels']
+            else:
+                return False
+
+            if isinstance(labels, basestring):
+                labels = yaml.safe_load(labels)
+            if not isinstance(labels, dict):
+                raise errors.AnsibleFilterError(
+                    "failed expected node labels to be a dict or serializable to a dict"
+                )
+            return label in labels and (value is None or labels[label] == value)
```

Comment:
If called without a value specified, it will return all nodes that have the label.  Will add a comment to clarify.